### ISSUE TITLE
Concurrent mode

### ISSUE THREAD

**Issue by dminkovsky** [4Catalyzer/found#638]
I am checking out [relay@experimental](https://relay.dev/docs/en/experimental/a-guided-tour-of-relay) with Suspense/Concurrent Mode and am experiencing some issues with `[useTransition](https://reactjs.org/docs/concurrent-mode-reference.html#usetransition)`. It seems like `useTransition({timeoutMs})` is not being effective for suppressing updates, and I am guessing that's because this library uses Redux for state management (vs. `setState()`). Has anyone experienced this? If not, I could create a sandbox that demonstrates the issue.

**Comment by taion** [comment]
I haven't tried this yet.

My plan FWIW is to refactor this library quite a bit for concurrent mode – much of the complexity here is in rendering various loading/pending states, essentially as a user-space implementation of the functionality that suspense now provides natively.

As these problems now get handled by React proper, it seems like it would be easier to work with things if that support were factored out.

**Comment by dminkovsky** [comment]
Yes, I went through that exercise when I was playing with this and it was pretty fun: very little was left, concurrent mode really does handle all the heavy lifting. Looking forward to this version.

**Comment by taion** [comment]
I switched jobs a few years ago and I no longer actively use or maintain this package, sorry. This would have to be a question for whoever the current maintainer is.
